tensorflow/lite/kernels/activations.cc

```diff
@@ -426,13 +425,19 @@ TfLiteStatus TanhPrepare(TfLiteContext* context, TfLiteNode* node) {
         (data->input_left_shift == 0 || data->input_left_shift == 1);
 
     if (!param_scale_pot) {
-      // In case of general scale parameter, we need to do a rescaling.
-      // Magic constant 4096:
-      // We need to scale down to (-2^3, 2^3) / 3 is kInputIntegerBits/ interval
-      // from 16-bit (-2^15, 2^15),
-      // so we need to multiply by
-      // 2^(15 - kInputIntegerBits) = 2^12 = 4096.
-      data->input_multiplier = static_cast<int32_t>(input->params.scale * 4096);
+      // Calculate multiplier to change input scale to 1/(3*4096)
+      // as required by the table lookup.
+      // In this scaling +/-2^17 represents +/-10.7
+
+      double multiplier = input->params.scale * 4096.0 * 3.0;
+      data->input_left_shift = 0;
+
+      while (multiplier <= 32767.0 / 2.0 && data->input_left_shift <= 30) {
```

Comment:
looks like this can be replaced with a std::frexp.

Reply:
As @renjie-liu suggested, could this be replaced by `std::frexp` ?
(something like `multiplier =  std::frexp(multiplier, &data->input_left_shift)` ?)

This also applies to same logic at line 550

Reply:
Hi @fredrec Sorry for the late reply.
This block can't be replaced with this function, because frexp breaks in double significant and exponent, but we need int16 here.
for example,
4.125 should be 
multiplier: 16896
input left shift: 12
but if I apply frexp, then the result is 0.515625 and 3.

Thanks for the review!